src/libs/Tokenizer.js

```diff
@@ -0,0 +1,95 @@
+
+
+var theTokenizer;
+
+class Tokenizer {
+
+    constructor(program, literalsList) {
+        this.program = program;
+        this.literals = literalsList;
+        this.tokens = [];
+        this.currentToken = 0;
+        this.tokenize();
+    }
+
+    tokenize() {
+        let tokenizedProgram = this.program;
+        tokenizedProgram = tokenizedProgram.replace(/\n/g, "");
+
+        // extracts replacing them with _$_, stores them in an array
+        // to be added back in later.
+        var texts = [...tokenizedProgram.matchAll(/".*"/g)];
+        tokenizedProgram = tokenizedProgram.replace(/".*"/g, "_$_");
+
+        // prevents leading 'lat' in lat lon from combining with variable name token
+        tokenizedProgram = tokenizedProgram.replace(/( \d+)/g, function(m){
```

Comment:
How could they ever be combined with the variable name? The user must provide a space between the variable name and the LAT value and also, your implementation actually depends on this assumption (the regex starts with a single white space)?

Reply:
Yeah in this implementation (following Elisa's) we actually strip out all the spaces. So this merges the tokens of lat and variable name.

Reply:
I think there's another version of the same function that doesn't strip out the spaces...but not sure if we want to do that

Reply:
@patrickdhopkins we'll have to remove this: the regular expression does not cover the case when latitude is negative. Also, we don't need to add the the separator after you added the code for adding the underscores surrounding numerical values (the replace method that uses regex `/[-]*\d+\.\d+/g`). I will merge the PR so we can continue working on it.